aten/src/ATen/native/cuda/Activation.cpp

```diff
@@ -90,7 +90,7 @@ std::tuple<Tensor&, Tensor&> log_sigmoid_forward_out_cuda(const Tensor& input, T
 
 std::tuple<Tensor, Tensor> log_sigmoid_forward_cuda(const Tensor& input) {
   auto result = at::empty_like(input);
-  auto buffer = at::empty({0}, input.options());
+  auto buffer = at::empty_like(input);
```

Comment:
This is hardcoded as a rank 1 tensor! and that's an issue because log_sigmoid_backward has a pointwise boxed batching rule which promotes the rest of the scalar arguments to rank 1. 

This changing the shape here is OK because if you look at `log_sigmoid_forward_out_cuda` it's actually unused. (It is only used on CPU and CPU/CUDA need to have the same signature)

Reply:
There is a little more overhead now as we must allocate a scalar zero-valued tensor instead of a empty zero-numel tensor on cuda. 

Why?
- We can no longer return a zero-element tensor because that a tensor needs to be at least rank 1 to have zero numel. 
- test_decomp needs to compare the outputs, so we use zeros instead of empty

Alternatives:
- add a special skip for log_sigmoid_forward in the decomp test and return an empty scalar tensor
- maybe vmap's point-wise batching rule should not promote if the tensor is zero-element

Reply:
Is the problem that `buffer` is a different shape depending on whether or not we go down the CPU or CUDA path? And the batching rule works in one of the cases (the cpu path) but not the CUDA path, because the CUDA path hardcodes the buffer to be a certain shape?

Reply:
Yup

Reply:
The least disruptive thing we can do here is write a new batching rule for log_sigmoid_forward that handles the CPU/CUDA cases. The current batching rule isn't correct, as you noted.

Reply:
cc @ngimel for another opinion on performance. Do you think is it OK for a PyTorch operator to return Tensors with different shape depending on if it runs on CPU or CUDA? (in this PR, this happens in the case of log_sigmoid_forward). If so, that is what this PR attempts to fix, but making the output of log_sigmoid_forward have the same shape on CUDA involves adding an additional Tensor allocation, which may impact performance